libpod/options.go

```diff
@@ -1339,17 +1340,46 @@ func WithCommand(command []string) CtrCreateOption {
 // This creates a container from a directory on disk and not an image.
 func WithRootFS(rootfs string) CtrCreateOption {
 	return func(ctr *Container) error {
+		var err error
 		if ctr.valid {
 			return define.ErrCtrFinalized
 		}
-		if _, err := os.Stat(rootfs); err != nil {
+
+		rootfs, err = withRootFSOverlay(ctr, rootfs)
+		if err != nil {
+			return err
+		}
+		if _, err = os.Stat(rootfs); err != nil {
 			return err
 		}
+
 		ctr.config.Rootfs = rootfs
 		return nil
 	}
 }
 
+func withRootFSOverlay(ctr *Container, rootfs string) (string, error) {
+	// Note: This is just an experiment and will be more generic in final draft.
+	parts := strings.SplitN(rootfs, ":", 2)
+	// lookout for --rootfs /some/path:O
+	if len(parts) > 1 && parts[1] == "O" {
+		overlaySrc := parts[0]
+		// remove tmp from here and take value of --tmpfs from cli
+		overlayDest := ctr.runtime.store.GraphRoot()
+		contentDir, err := overlay.GenerateStructure(ctr.runtime.store.GraphRoot(), ctr.ID(), "rootfs", ctr.RootUID(), ctr.RootGID())
+		if err != nil {
+			return "", errors.Wrapf(err, "rootfs-overlay: failed to create TempDir in the %s directory", overlayDest)
+		}
+		overlayMount, err := overlay.Mount(contentDir, overlaySrc, overlayDest, ctr.RootUID(), ctr.RootGID(), ctr.runtime.store.GraphOptions())
```

Comment:
This doesn't seem right - are we keeping the overlay mounted continuously? Can't we mount it only when the container is running? What happens if the system reboots and the FS is unmounted?

Reply:
@mheon It is there as long as the container exists since there could be cases were stuff is written on overlay mount that was the original idea to keep mount intact and let it cleaned up as soon as container is removed.

Reply:
That doesn't sound right - we don't like to leave mounts around, taking up resources on the host system, when containers are not running. It seems more correct to move this into container mount / unmount logic, where it will act like other containers do.

Reply:
@mheon Inoder to expect rootfs as overlay this has to be mounted before container is invoked. This might not follow the path of regular mounts but I'll add changes to perform cleanups.

Reply:
Sure - every container mounts before starting, as part of the normal mount process. We should be able to hook the existing `mountStorage()` function to mount the overlay. Or is there some other limitations here I'm not aware of?

Reply:
this is resolved and moved to `mountStorage(`